### DISCUSSION TITLE
Strange Image behavior when filtering rows

### DISCUSSION THREAD

**Discussion by soutieslak** [exceljs/exceljs#1913] [Q&A] [answered]
I noticed that if you have multiple rows with images, and you use excels "filter" feature, then there was some weird behavior with the images.

Upon further investigation, I noticed that there is a property set on the images by default:

Right Click Image
Select 'Size and Properties'
Under Properties there are 3 options:
- 'Move and size with cells'
- 'Move but don't size with cells' <- This is the default
- 'Don't move or size with cells'

I noticed that using the option 'Move and size with cells' fixes this issue when you filter through lots of rows.

I've possibly found the answer:

![image](https://user-images.githubusercontent.com/5350054/143891800-ed4cd7cb-e8dc-4787-ae5b-b79459d16dce.png)

But the issue now is, how do you specify where the image must go e.g I want the image in Cell A1, so I specify col 1, row 1 for the 'top left'(tl) and 'bottom right' (br)

{
              "tl": { "col": 1, "row": 1},
              "br": { "col": 1, "row": 1},
              "editAs" : "undefined"
};

but its not working... What is the correct way to specify tl and br if I want to refer to a cell?

**Comment by soutieslak** [comment, marked as answer]
I figured it out, this worked! Whoop!

            let imgProperties = {
              "tl": { "col": imageColumn - 1, "row": sheet[0].rowCount - 1},
              "br": { "col": imageColumn, "row": sheet[0].rowCount},
              "editAs" : "undefined"
              };
              
            sheet[0].addImage(imageId1, imgProperties);

**Comment by StefanV85** [comment]
Hello, 

i tried your code, but it leads in my opinion to an corrupted excel document. Excel App always ask me, if it should repair the document.

=> The Reason is, that the value "undefined" will be placed in the xml document of the excel. and "undefined" is not a supported value for anchor types.

[https://github.com/exceljs/exceljs/issues/1455](https://github.com/exceljs/exceljs/issues/1455)
reference to the ms office documentation. 

"twoCell" is the correct value in my opinion. The documentation is not correct here.

here is my Code: 
`

  const workbook = new Excel.Workbook();
  // await workbook.xlsx.readFile("./template2.xlsx");
  let images: number[] = [];
  workbook.addWorksheet("Charts");
  const imageId1 = workbook.addImage({
    filename: "./Logo.jpg",
    extension: "jpeg",
  });
  const imageId2 = workbook.addImage({
    filename: "./Logo2.jpg",
    extension: "jpeg",
  });
  images.push(imageId1);
  images.push(imageId2);

  let worksheet = workbook.worksheets[0];

  let columnIndex = 0;
  for (let rowIndex = 0; rowIndex < images.length; rowIndex++) {
    worksheet.addImage(images[rowIndex], {
      // @ts-expect-error Issue with ExcelJs types.
      tl: {
        col: columnIndex,
        row: rowIndex,
        nativeCol: columnIndex,
        nativeRowOff: 0,
        nativeRow: rowIndex,
        nativeColOff: 0,
        worksheet: worksheet,
        colWidth: 50,
        rowHeight: 50,
      },
      // @ts-expect-error Issue with ExcelJs types.
      br: {
        col: columnIndex + 1,
        row: rowIndex + 1,
        nativeCol: columnIndex + 1,
        nativeRowOff: 0,
        nativeRow: rowIndex + 1,
        nativeColOff: 0,
        worksheet: worksheet,
        colWidth: 50,
        rowHeight: 50,
      },
      editAs: "twoCell",
    });

`

Kind Regards 
Stefan
